### ISSUE TITLE
Setup coupled WAM-IPE(Refactored) runs

### ISSUE THREAD

**Issue by joeschoonover** [SWPC-IPE/WAM-IPE#260]
Now that the coupled WAM-IPE (refactored) code compiles, the scripts used to configure and provision a model run need to be updated to carry in the appropriate initial conditions, grid, forcing, and namelist files for the refactored IPE.

Conditions for Satisfaction : 
* WAM-IPE coupled run should be completed for 24 hrs without stopping - note that plasma fields may not be correct due to #251

* In recognition that #251 is currently unresolved, we should verify that the WAM neutral fields are properly being passed into IPE.

* We should verify that time varying and fixed drivers are functioning as expected.

**Comment by joeschoonover** [comment]
This issue overlaps with #236

**Comment by joeschoonover** [comment]
Currently, WAM-IPE coupled fails. The likely cause is the use of conflicting hdf5 libraries at compile time.

* IPE depends on hdf5parallel/1.10.4 ( compiled with intel/16.1.150 and impi/5.1.2.150 ) and netcdf/4.3.0 ( compiled with intel/16.1.150 ). It's unclear what version of HDF5 was used to build this version of NetCDF. For those unfamiliar, NetCDF is built on top of HDF5.

* ESMF appears to depend on netcdf/4.3.0 and impi/5.1.1.109

At compile time, we see the warning : 
```
ld: warning: libhdf5_hl.so.9, needed by /apps/netcdf/4.3.0-intel/lib/libnetcdf.so, may conflict with libhdf5_hl.so.100
ld: warning: libhdf5.so.9, needed by /apps/netcdf/4.3.0-intel/lib/libnetcdf.so, may conflict with libhdf5.so.103
```
suggesting that the hdf5parallel library has different libraries associated with it than the hdf5 build that was used to build netcdf.

Additionally, at runtime, HDF5 throws the following warning : 
```
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.8.14, library is 1.10.4
```
Further, the crash ultimately happens at [line 803 of the `IPELIB/src/IPE_Grid_Class.F90` source code](https://github.com/SWPC-IPE/WAM-IPE/blob/ipe_refactor/IPELIB/src/IPE_Grid_Class.F90#L803). This line corresponds to the opening of the `IPE_Grid.nc` netcdf file.


This message is consistent with the idea that the netcdf/4.3.0 libraries were built with hdf5/1.8.14 . If this idea is true, then swapping netcdf/4.3.0 with a netcdf library built with hdf5/1.10.4 would remedy the issue.

Theia does offer `netcdf-hdf5parallel/4.4.0` package; in fact, this is the netcdf package we use to build IPE standalone. I'll attempt to make the swap in `modulefiles/theia/wam-ipe` from the `netcdf/4.3.0` module to `netcdf-hdf5parallel/4.4.0`

**Comment by rmontuoro** [comment]
@joeschoonover, both `esmf/7.1.0r` and `esmf/7.1.0rp1bs01` (the ESMF version currently used in NEMSfv3gfs and other apps) depend on the `netcdf` module but not on `netcdf-hdf5parallel/4.4.0` on Theia.
I'm not aware of any other ESMF build that uses the `netcdf-hdf5parallel/4.4.0`.

**Comment by joeschoonover** [comment]
Is there a way to build ESMF using `netcdf-parallel/4.4.0` on theia ?

**Comment by rmontuoro** [comment]
We can build a custom ESMF library for WAM-IPE on Theia but I would recommend using what's being used in NEMSfv3gfs at the moment, since this is the vetted version of ESMF used by all other NEMS apps. Can we use `pnetcdf` instead of `netcdf-parallel/4.4.0`?

**Comment by joeschoonover** [comment]
With the new modules in place, MPI_Abort is called within GSM when running coupled WAM-IPE. To reproduce the error, download a fresh copy of the ipe_refactor branch and do
```
NEMS/NEMSAppBuilder rebuild app=wam-ipe
cd scripts/compsets
./submit.sh march2013_regression.config
```
The `fcst` file will show the error message.

**Comment by oehmke** [comment]
Hi Joe, 

  Does it seem to be ESMF related? Is there an error in the Pet*.LogFiles? 

Thanks,

- Bob

**Comment by joeschoonover** [comment]
Good call. I checked one of the PET files corresponding to a failed rank, and it shows
```
20181207 160502.178 ERROR            PET49 ESMF_Time.F90:810 ESMF_TimeGet() Object Set or SetDefault method not called  - Object not Initialized
20181207 160502.178 ERROR            PET49 IPE_Wrapper.F90:66 Object Set or SetDefault method not called  - Passing error in return code
20181207 160502.178 ERROR            PET49 ipeCap.F90:262 Object Set or SetDefault method not called  - Passing error in return code
20181207 160502.178 ERROR            PET49 EARTH grid component01:src/addon/NUOPC/src/NUOPC_Driver.F90:1529 Object Set or SetDefault method not called  - Phase 'IPDv02p3' Initialize for modelComp 3: IPM did not return ESMF_SUCCESS
20181207 160502.178 ERROR            PET49 EARTH grid component01:src/addon/NUOPC/src/NUOPC_Driver.F90:1189 Object Set or SetDefault method not called  - Passing error in return code
20181207 160502.178 ERROR            PET49 EARTH grid component01:src/addon/NUOPC/src/NUOPC_Driver.F90:313 Object Set or SetDefault method not called  - Passing error in return code
20181207 160502.178 ERROR            PET49 module_NEMS_GRID_COMP.F90:666 Object Set or SetDefault method not called  - Breaking out of subroutine
20181207 160502.179 ERROR            PET49 MAIN_NEMS.F90:466 Object Set or SetDefault method not called  - Aborting NEMS
20181207 160502.179 INFO             PET49 Finalizing ESMF
                                                             
```
After checking a few more of the PET files corresponding to the failed ranks, I find the same issue. This looks like something I didn't quite get right in the IPE_Wrapper. I'll do some comparisons with older code to make sure everything is in place.

If you have guidance on how to fix this, @oehmke , that would be very helpful

**Comment by joeschoonover** [comment]
At build time, this warning appears :
```
ld: warning: libmpi.so.4, needed by /apps/hdf5/1.8.14-intel-impi/lib/libhdf5hl_fortran.so, may conflict with libmpi.so.12
```
I'm not sure if it causes any significant issues

**Comment by joeschoonover** [comment]
The first draft of the plotting utilities are complete on the ipe_refactor branch. To generate plots, use `scripts/plots/make_plots.batch`.
```
cd scripts/plots
```
Modify `make_plots.batch` to point `IPE_RUNDIR` to the directory where IPE hdf5 files can be found. Then, run
```
./make_plots.batch
```
This creates a job in the debug queue to convert HDF5-Apex files to NetCDF-geographic files and runs the `plot.py` plotting utility.

You can add plots, or change the appearance of plots under `scripts/plots/plot_settings.yaml`. If you would like to add new plots or recommend changes in the plot aesthetics, make changes to `plot_settings.yaml` on a feature branch and submit a pull request

**Comment by joeschoonover** [comment]
Plots from a 24 hr run of coupled WAM-IPE (Refactored) can be found [here](https://drive.google.com/drive/folders/17YbyHDONAQSkTJiWL2LThwVLtn2Fy6Zh?usp=sharing)

Note that this run is meant only to test the passing of WAM variables to IPE. Because of this, we should only focus on the neutral atmosphere state perceived by IPE.

**Comment by joeschoonover** [comment]
This run was conducted at fixed f10.7 of 120 and kp of 3

**Comment by twfang** [comment]
I think the fields look good. A few comments for these plots. 

1. Several fields are not correct at 0UT. Perhaps you can plot results from first time-step (3 min for WAM) or just not to include them. BTW, are there neutrals coming into IPE within the first 3 mins?

2. Colors for winds should be reversed (i.e., red for eastward/northward/upward). Just the way how people are used to. 
3. Color scales for oxygen needs to be changed.
